pkgs/os-specific/linux/minimal-bootstrap/default.nix

```diff
@@ -0,0 +1,24 @@
+{ lib
```

Comment:
Regarding the filesystem layout: is this really *Linux-specific*?

I mean obviously that's going to be the first platform supported, but at that point how hard is it to do the same thing on a different kernel?  Surely nothing in here assumes glibc, right?

I guess I should wait to see the finished result, but if you pull it off `pkgs/stdenv/mes` might be a better place to put it.

Reply:
Hrm, it looks like the bootstrap chain at some point [builds a Linux kernel](https://man.sr.ht/~oriansj/bootstrappable/live-bootstrap.md#linux-4910) and then [`kexec`s into it](https://man.sr.ht/~oriansj/bootstrappable/live-bootstrap.md#kexec-tools-2022).

This happens *before* the [first copy of gcc](https://man.sr.ht/~oriansj/bootstrappable/live-bootstrap.md#gcc-474) is built.

Is this really required?  Obviously `kexec` does not work (at least I hope not!) in the Nix sandbox.  And dragging in `qemu` would mean that it isn't really a bootstrap anymore (since you need `gcc` to build `qemu`).

I've heard crazy schemes about running kaem, M2-Planet, and all the rest on bare metal straight from the reset vector (like coreboot, or the "pre-EFI" blobs on non-coreboot systems).  That's very cool, and those people will want to do it this way.  But hopefully we can get a GCC built without having to do Linux first...

Reply:
live-bootstrap is where we can draw a lot of inspiration (and build scripts!) but it's a project with different goals. We won't have to do something as complicated as building a kernel!

Reply:
I not the OP, but I do see that mes copies `lib/linux/x86-mes-gcc/crt*.c` from its distribution.

Reply:
My guess is the hand-coded ELFs might be linux specific. We can move things around later.